tests/test_metrics/test_lvis_detection_metric.py

```diff
@@ -0,0 +1,400 @@
+        {
+            'proposal_nums': 30
+        },
+    ])
+def test_segm_metric_interface(metric_kwargs):
+    tmp_dir = tempfile.TemporaryDirectory()
+
+    # create dummy data
+    fake_json_file = osp.join(tmp_dir.name, 'fake_data.json')
+    _create_dummy_coco_json(fake_json_file)
+
+    num_classes = 2
+    metric = ['segm']
+    # Avoid some potential error
+    fake_dataset_metas = {
+        'classes': tuple([str(i) for i in range(num_classes)])
+    }
+    coco_det_metric = LVISDetection(
+        ann_file=fake_json_file,
+        metric=metric,
+        dataset_meta=fake_dataset_metas,
+        **metric_kwargs)
+    assert isinstance(coco_det_metric, BaseMetric)
+
+    metric_results = coco_det_metric(
+        predictions=[_create_dummy_results()],
+        groundtruths=[dict()],
+    )
+    assert isinstance(metric_results, dict)
+    assert 'segm_AP' in metric_results
+    tmp_dir.cleanup()
+
+
+@pytest.mark.skipif(
+    coco_wrapper is None, reason='coco_wrapper is not available!')
+def test_metric_invalid_usage():
+    tmp_dir = tempfile.TemporaryDirectory()
+
+    # create dummy data
+    fake_json_file = osp.join(tmp_dir.name, 'fake_data.json')
+    _create_dummy_coco_json(fake_json_file)
+
+    with pytest.raises(KeyError):
+        LVISDetection(ann_file=fake_json_file, metric='xxx')
+
+    with pytest.raises(TypeError):
+        LVISDetection(ann_file=fake_json_file, iou_thrs=1)
+
+    with pytest.raises(AssertionError):
+        LVISDetection(ann_file=fake_json_file, format_only=True)
+
+    num_classes = 2
+    # Avoid some potential error
+    fake_dataset_metas = {
+        'classes': tuple([str(i) for i in range(num_classes)])
+    }
+    coco_det_metric = LVISDetection(
+        ann_file=fake_json_file, dataset_meta=fake_dataset_metas)
+
+    with pytest.raises(KeyError):
+        prediction = _gen_prediction(num_classes=num_classes)
+        groundtruth = _gen_groundtruth(num_classes=num_classes)
+        del prediction['bboxes']
+        coco_det_metric([prediction], [groundtruth])
+
+    with pytest.raises(AssertionError):
+        prediction = _gen_prediction(num_classes=num_classes)
+        groundtruth = _gen_groundtruth(num_classes=num_classes)
+        coco_det_metric(prediction, groundtruth)
+    tmp_dir.cleanup()
+
+
+@pytest.mark.skipif(
+    coco_wrapper is None, reason='coco_wrapper is not available!')
+def test_compute_metric():
+    tmp_dir = tempfile.TemporaryDirectory()
+
+    # create dummy data
+    fake_json_file = osp.join(tmp_dir.name, 'fake_data.json')
+    _create_dummy_coco_json(fake_json_file)
+    dummy_pred = _create_dummy_results()
+    fake_dataset_metas = dict(classes=['car', 'bicycle'])
+
+    # test single coco dataset evaluation
+    coco_det_metric = LVISDetection(
+        ann_file=fake_json_file,
+        classwise=False,
+        outfile_prefix=f'{tmp_dir.name}/test',
+        dataset_meta=fake_dataset_metas)
+    eval_results = coco_det_metric([dummy_pred], [dict()])
```

Comment:
There should be 1 more test case: call `metric.add` >=2 times and then `metric.compute`
This covers a basic use case.